### PR TITLE
generic parsers

### PR DESCRIPTION
**Pull request by lockwooddev**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,41 @@
...
+
+### Rendering [Bandit](https://github.com/PyCQA/bandit) output for Slack
+
+```yaml
+- name: Render Bandit output for slack
+  id: bandit-to-slack
+  uses: cbrix-company/action-messages@v1
+  with:
+    tool: bandit
+    input-file: 'bandit_output.json'
```
Comment (k-shlomi) [inline README.md:23]: are we missing output-file definition here too?

File: main.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+registry = {
+    'bandit': BanditMessageRenderer,
+}
+
+
+def load_input(file_path):
+    with open(file_path, 'r') as f_:
+        if file_path.endswith('.json'):
```
Comment (k-shlomi) [inline main.py]: don't open the file if that's the case.. check it before

File: entrypoint.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh -l
+context='{"repository":"$1","ref":"$2","actor":"$3","run_id":"$4","sha":"$5"}'
```
Comment (k-shlomi) [inline entrypoint.sh]: is there a way to name these?

File: renderers/bandit.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
+import json
+
+from .base import BaseMessageRenderer
+
+
+class BanditMessageRenderer(BaseMessageRenderer):
+    """
+    Renders top 2 bandit findings
```
Comment (k-shlomi) [inline renderers/bandit.py:8]: shouldn't this be configurable?

File: renderers/bandit.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+        results = []
+        for idx, result in enumerate(self.input['results']):
+            issue_severity = result['issue_severity']
+            distribution[issue_severity] += 1
+            issue_confidence = result['issue_confidence']
+            results.append(
+                (
+                    idx,
+                    self.__class__.LEVELS[issue_severity],
```
Comment (k-shlomi) [inline renderers/bandit.py:43]: I think `__class__` is not needed here. just `self.LEVELS`

File: renderers/bandit.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                    self.__class__.LEVELS[issue_confidence],
```
Comment (k-shlomi) [inline renderers/bandit.py:44]: confidence could be undefined?

File: renderers/bandit.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+                    result['line_number'],
+                    issue_severity,
+                )
+            )
+
+        sorted_by_severity = sorted(results, key=lambda x: x[1], reverse=True)
+        sorted_by_confidence = sorted(sorted_by_severity, key=lambda x: x[2], reverse=True)
+
+        return {
+            'top_issues': sorted_by_confidence[:2],
```
Comment (k-shlomi) [inline renderers/bandit.py:57]: hard coded 2 here is weird..

File: renderers/bandit.py (hunk 4 above)
Comment (k-shlomi) [reply renderers/bandit.py:8]: top 2 by confidence.. not severity.. should probably be documented

File: renderers/bandit.py [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+    def render_to_slack(self):
+        repo_name = self.context['repository']
+        distribution = self.context['distribution']
+        runtime = self.context['runtime']
+
+        high = distribution["HIGH"]
+        medium = distribution["MEDIUM"]
+        low = distribution["LOW"]
+        header = (
```
Comment (k-shlomi) [inline renderers/bandit.py]: don't define this inside a function.

**Review by k-shlomi — COMMENTED** [review]
CR

File: README.md (hunk 1 above)
Comment (lockwooddev) [reply README.md:23]: Yes, but I fixed that

File: entrypoint.sh (hunk 3 above)
Comment (lockwooddev) [reply entrypoint.sh]: Not possible

File: renderers/bandit.py (hunk 4 above)
Comment (lockwooddev) [reply renderers/bandit.py:8]: For now it is not configurable. That would also make the action a lot more complicated and would possibly mean that we would have to split per security input.

File: renderers/bandit.py (hunk 5 above)
Comment (lockwooddev) [reply renderers/bandit.py:43]: Both work, but `self.LEVELS` implies getting it from the instance.

File: renderers/bandit.py (hunk 6 above)
Comment (lockwooddev) [reply renderers/bandit.py:44]: Yes it is a possibility in the output of bandit

File: renderers/bandit.py (hunk 7 above)
Comment (lockwooddev) [reply renderers/bandit.py:57]: It really depends if we want to configure this all the way through the action. We would have to implement parameters for each renderer in the action.yaml.

**Review by k-shlomi — APPROVED** [review]

**Pull request merged**